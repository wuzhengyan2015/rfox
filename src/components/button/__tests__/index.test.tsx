import { render, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import Button from '../index'

configure({ adapter: new Adapter() })

describe('Button tests', () => {
    it('render correctly', () => {
        const treeOne = render(
            <Button type='primary' size="large" onClick={() => {}} shape="round">Button</Button>
        )
        const treeTwo = render(
            <Button href="http://www.baidu.com" target="_blank">Link</Button>
        )
        expect(treeOne).toMatchSnapshot()
        expect(treeTwo).toMatchSnapshot()
    })
 })