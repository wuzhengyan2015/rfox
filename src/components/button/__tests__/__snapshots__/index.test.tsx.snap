// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests render correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "rfox-btn rfox-btn-primary rfox-btn-lg rfox-btn-round",
      "type": "button",
    },
    "children": Array [
      Object {
        "data": "Button",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Button tests render correctly 2`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "rfox-btn",
      "href": "http://www.baidu.com",
      "target": "_blank",
    },
    "children": Array [
      Object {
        "data": "Link",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "a",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "href": undefined,
      "target": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "href": undefined,
      "target": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
